src/data/mv_polynomial/monad.lean

```diff
@@ -291,7 +291,7 @@ lemma eval₂_hom_comp_bind₂ (f : S →+* T) (g : σ → T) (h : R →+* mv_po
 by { ext1, apply eval₂_hom_bind₂ }
 
 lemma aeval_bind₂ [algebra S T] (f : σ → T) (g : R →+* mv_polynomial σ S) (φ : mv_polynomial σ R) :
-  aeval f (bind₂ g φ) = eval₂_hom ((@aeval σ S T f _ _ _ : mv_polynomial σ S →+* T).comp g) f φ :=
+  aeval f (bind₂ g φ) = eval₂_hom ((@aeval S σ _ T f _ _ : mv_polynomial σ S →+* T).comp g) f φ :=
```

Comment:
I'm not happy about this (and suspect a lot more has been broken). The order of the arguments has been changed in `aeval`.

Reply:
Right... this is exactly why we would like to have all the variables renamed, because they were used with very different meanings in different declarations.